hepstats/hypotests/calculators/frequentist_calculator.py

```diff
@@ -0,0 +1,424 @@
+from ..fitutils.utils import pll, get_nevents
+from ..fitutils.sampling import base_sampler, base_sample
+from ..parameters import POI, POIarray
+
+
+class FrequentistCalculator(BaseCalculator):
+    """
+    Class for frequentist calculators.
+    """
+
+    def __init__(self, input, minimizer, ntoysnull=100, ntoysalt=100, sampler=base_sampler, sample=base_sample):
+        """Frequentist calculator class.
+
+            Args:
+                input : loss or fit result
+                minimizer : minimizer to use to find the minimum of the loss function
+                ntoysnull (int, default=100): number of toys to generate for the null hypothesis
+                ntoysalt (int, default=100): number of toys to generate for the alternative hypothesis
+                sampler : function used to create sampler with models, number of events and
+                    floating parameters in the sample Default is `hepstats.fitutils.sampling.base_sampler`.
+                sample : function used to get samples from the sampler.
+                    Default is `hepstats.fitutils.sampling.base_sample`.
+
+            Example with `zfit`:
+                >>> import zfit
+                >>> from zfit.core.loss import UnbinnedNLL
+                >>> from zfit.minimize import MinuitMinimizer
+
+                >>> obs = zfit.Space('x', limits=(0.1, 2.0))
+                >>> data = zfit.data.Data.from_numpy(obs=obs, array=np.random.normal(1.2, 0.1, 10000))
+                >>> mean = zfit.Parameter("mu", 1.2)
+                >>> sigma = zfit.Parameter("sigma", 0.1)
+                >>> model = zfit.pdf.Gauss(obs=obs, mu=mean, sigma=sigma)
+                >>> loss = UnbinnedNLL(model=[model], data=[data], fit_range=[obs])
+
+                >>> calc = FrequentistCalculator(input=loss, minimizer=MinuitMinimizer(), ntoysnull=1000, ntoysalt=1000)
+        """
+
+        super(FrequentistCalculator, self).__init__(input, minimizer)
+
+        self._toysnull = {}
+        self._toysalt = {}
+        self._ntoysnull = ntoysnull
+        self._ntoysalt = ntoysalt
+        self._sampler = sampler
+        self._sample = sample
+        self._toys_loss = {}
+
+    @property
+    def ntoysnull(self):
+        """
+        Returns the number of toys loss for the null hypothesis.
+        """
+        return self._ntoysnull
+
+    @property
+    def ntoysalt(self):
+        """
+        Returns the number of toys loss for the alternative hypothesis.
+        """
+        return self._ntoysalt
+
+    def sampler(self, floating_params=None, *args, **kwargs):
+        """
+        Create sampler with models.
+
+        Args:
+            floating_params (list): floating parameters in the sampler
+
+        Example with `zfit`:
+            >>> sampler = calc.sampler(floating_params=[zfit.Parameter("mean")])
+        """
+        self.set_dependents_to_bestfit()
+        nevents = []
+        for m, d in zip(self.model, self.data):
```

Comment:
This line puzzled me for a sec because none of `self.model` and `self.data` are defined here. I did not check but they must be defined in the `BaseCalculator`, right? Why not write a little comment for this for the non-experts in the package ;-)?

Reply:
@eduardo-rodrigues  yeah indeed `model` and `data` are defined in `BaseCalculator`. I will add a line.